messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -255,4 +292,60 @@ protected void createCheckpoint() {
       currentCheckpoint = new CheckpointList(partitionMinList);
     }
   }
+
+  @Override
+  public synchronized void close() {
+    newReaders.clear();
+    if (null != partitionDiscovererAndReaderCreator) {
+      partitionDiscovererAndReaderCreator.cancel();
+    }
+    super.close();
+  }
+
+  protected void startNewReaders() throws IOException{
+    for (PartitionId id : newReaders.keySet()) {
+      PartitionReader reader = newReaders.get(id);
+      reader.start(getReaderNameSuffix());
+      readers.put(id, reader);
+      LOG.info("started new reader " + getReaderNameSuffix() +
+          " " + reader.toString());
+    }
+    newReaders.clear();
+  }
+
+  protected void doReset() throws IOException {
+    initDone = false;
+    super.doReset();
+    if (streamType.equals(StreamType.COLLECTOR)) {
+      createAndScheduleDiscovererTimer();
+    }
+  }
+
+  private void createAndScheduleDiscovererTimer() {
+    partitionDiscovererAndReaderCreator = new Timer(
+        "PartitionDiscovererAndReaderCreator");
+    scheduleTimer(partitionDiscovererAndReaderCreator);
+    LOG.info("Initialised timer for discovery of new collectors" +
+        " output with frequency " + discoverFrequency);
+  }
+
+  private void scheduleTimer(Timer partitionDiscovererAndReaderCreator) {
+    partitionDiscovererAndReaderCreator.schedule(new TimerTask() {
+          @Override
+          public void run() {
+            try {
+              if (initDone) {
+                LOG.info("collector discoverer activated");
+                createPartitionReaders();
+                LOG.info("Start new readers activated");
```

Comment:
this will help in giving the info in timer related things. whether the timer is running or not. whether it is getting triggered at right time or not etc.